src/chain_table.rs

```diff
@@ -21,121 +21,154 @@ use std::{
+    pub fn as_blocks(&self) -> &[Block<BLOCK_LENGTH>] {
+        &self.blocks[..self.length]
     }
 
     /// Encrypts the Chain Table value using the given DEM key.
     ///
     /// - `kwi_value`   : DEM key used to encrypt the value
     /// - `rng`         : random number generator
-    pub fn encrypt<const TABLE_WIDTH: usize, const KEY_LENGTH: usize, DEM: Dem<KEY_LENGTH>>(
+    pub fn encrypt<const KEY_LENGTH: usize, DEM: Dem<KEY_LENGTH>>(
         &self,
         kwi_value: &DEM::Key,
         rng: &mut impl CryptoRngCore,
     ) -> Result<Vec<u8>, Error> {
-        let mut ser = Serializer::with_capacity(BLOCK_LENGTH * TABLE_WIDTH);
-        for block in self.iter() {
-            ser.write(block)?;
-        }
-        // Pad the line with empty blocks if needed.
-        let padding = Block::<BLOCK_LENGTH>::new_empty_block();
-        for _ in self.len()..TABLE_WIDTH {
-            ser.write_array(&padding)?;
-        }
-        DEM::encrypt(rng, kwi_value, &ser.finalize(), None).map_err(Error::from)
+        let bytes = self.try_to_bytes()?;
+        DEM::encrypt(rng, kwi_value, &bytes, None).map_err(Error::from)
     }
 
     /// Decrypts the Chain Table value using the given DEM key.
     ///
     /// - `kwi_value`   : DEM key used to encrypt the value
     /// - `ciphertext`  : encrypted Chain Table value
-    pub fn decrypt<
-        const TABLE_WIDTH: usize,
-        const DEM_KEY_LENGTH: usize,
-        DEM: Dem<DEM_KEY_LENGTH>,
-    >(
+    pub fn decrypt<const DEM_KEY_LENGTH: usize, DEM: Dem<DEM_KEY_LENGTH>>(
         kwi_value: &DEM::Key,
         ciphertext: &[u8],
     ) -> Result<Self, Error> {
-        if TABLE_WIDTH * BLOCK_LENGTH + DEM::ENCRYPTION_OVERHEAD != ciphertext.len() {
+        if 1 + TABLE_WIDTH * (1 + BLOCK_LENGTH) + DEM::ENCRYPTION_OVERHEAD != ciphertext.len() {
             return Err(Error::CryptoError(format!(
                 "invalid ciphertext length: given {}, should be {}",
                 ciphertext.len(),
-                TABLE_WIDTH * BLOCK_LENGTH + DEM::ENCRYPTION_OVERHEAD
+                1 + TABLE_WIDTH * (1 + BLOCK_LENGTH) + DEM::ENCRYPTION_OVERHEAD
             )));
         }
         let bytes = DEM::decrypt(kwi_value, ciphertext, None)?;
-        let mut de = Deserializer::new(&bytes);
-        let mut res = Vec::with_capacity(TABLE_WIDTH);
-        for _ in 0..TABLE_WIDTH {
-            let block = de.read::<Block<BLOCK_LENGTH>>()?;
-            // Blocks starting by `0` are padding.
-            // There cannot be any meaningful block after a padding block.
-            if block[0] == 0 {
-                break;
-            }
-            res.push(block);
-        }
-        Ok(Self(res))
+        Self::try_from_bytes(&bytes)
     }
 }
 
-impl<const BLOCK_LENGTH: usize> Deref for ChainTableValue<BLOCK_LENGTH> {
-    type Target = Vec<Block<BLOCK_LENGTH>>;
+impl<const TABLE_WIDTH: usize, const BLOCK_LENGTH: usize> Serializable
```

Comment:
No remark about this serialization (LGTM). But could you please update the README.md in order to detail the format of the 2 `index` tables. Ideally the structure (including the underlying structure of a `block` and the padding used) and the size of the chain table value should be given.

Also could you please add a table with some very basic examples of the expected size of these indexes (without graphs) when: 
- indexing 1 keyword gives an overhead of ... 
- indexing 2 keywords gives an overhead of ...
- ...

Something straightforward:

| Nb of words indexed | Nb of Entry Table Items | Nb of Chain Table Items | Size of Entry Table Item (in bytes) | Size of Chain Table Item (in bytes) | 
|---------------------|-------------------------|------------------------|----------------------------------|----------------------------------|
| 1                   | 1                       | 1                       | ...                      | ...                              | ...                 |
| 2                   | ...                       | ...                       | ...                      | ...                              | ...                 |

Reply:
do you think we could have a security reason why we have chosen TABLE_WIDTH = 5 and BLOCK_NUMBER=32 for the findex interfaces?

Reply:
done in 5fb568e